src/core/DataFile.cpp

```diff
@@ -901,6 +901,38 @@ void DataFile::upgrade_1_1_91()
 }
 
 
+void DataFile::upgrade_1_2_0_rc2_23()
+{
+	// Upgrade to version 1.1.92
+	QDomNodeList list;
+	QDomElement el;
+	QDomNode node;
+
+	// syncmode for multiple tempo sync knobs
+	list = elementsByTagName( "syncmode" );
+	for( int i = 0; !list.item( i ).isNull(); ++i )
+	{
+		int syncmode = el.attribute( "syncmode" ).toInt();
+		printf( "syncmode: %i", syncmode );
+		if( syncmode )
+		{
+			el = list.item( i ).toElement();
+			QDomNodeList syncgroup = el.parentNode().childNodes();
+			for( int i = 0; !syncgroup.item( i ).isNull(); ++i )
+			{
+				QString name = el.nodeName();
+				if( name.endsWith( "_numerator" ) )
+				{
+					name = name.remove( "_numerator" );
+					name = name.append( "_syncmode" );
+					el.setAttribute( name, syncmode );
```

Comment:
This should rather set only the last attribute "syncmode" in every element as only the last one get's saved currently. The other attributes wouldn't get written over as they are today.